src/config.rs

```diff
@@ -145,31 +145,55 @@ impl Builder {
         let toml_config: TomlConfig = toml::decode_str(&file_contents)
             .expect("unable to parse config file");
 
-        self.listen_ip = toml_config.server.listen_ip;
+        if let Some(table) = toml_config.server {
+            if let Some(val) = table.listen_ip {
+                self.listen_ip = val;
+            }
 
```

Comment:
Here, again, less whitespace would probably increase readability. The braces already create plenty of air.